clusterloader2/pkg/state/state.go

```diff
@@ -95,3 +102,20 @@ func (ns *NamespacesState) Delete(namespace string, identifier InstancesIdentifi
 	delete(namespaceState, identifier)
 	return nil
 }
+
+// ListAll returns a list of all registered instances in namespaces.
+func (ns *NamespacesState) ListAll() []InstanceTuple {
```

Comment:
Do you really need to transform map to list for what you're doing in the cleanup function?

Reply:
Yes/no/depends.
Yes - I want to list namespace, identifier and instances without giving access to the map. Cause interaction with state is done only using methods, implementation and structure of state can be changed any time (as long method signatures remain the same [we can introduce interface here])
No - I can return whole map. This way I can remove all methods and operate directly on structure.

IMHO first solution is better cause it has better encapsulation.
We can think about other way of iterating over items like iterating over namespaces and then for every namespace iterate over identifiers and then for every namespace, identifier getting instances. However, it looks like list approach is simpler.